src/ride/ride.c

```diff
@@ -995,7 +995,9 @@ void ride_remove_peeps(int rideIndex)
 			exitZ = ride->station_heights[stationIndex];
 			mapElement = ride_get_station_exit_element(ride, exitX, exitY, exitZ);
 
-			exitDirection = mapElement->type & 3;
+			if (mapElement != NULL) {
+				exitDirection = mapElement->type & 3;
+			}
```

Comment:
Yeah, I just realised that and issued a fix by clamping `exitDirection` regardless of it being set or not inside that `if`.